packages/astro/src/cli/preferences/index.ts

```diff
@@ -0,0 +1,150 @@
+		return listPreferences(settings, opts);
+	}
+
+	if (subcommand === 'enable' || subcommand === 'disable') {
+		key = `${key}.enabled` as PreferenceKey;
+	}
+
+	if (!isValidKey(key)) {
+		logger.error('preferences', `Unknown preference "${key}"\n`);
+		return 1;
+	}
+
+	if (subcommand === 'set' && value === undefined) {
+		const type = typeof dlv(DEFAULT_PREFERENCES, key);
+		// TODO: better error message
+		logger.error('preferences', `Please provide a ${type} value for "${key}"\n`);
+		return 1;
+	}
+
+	switch (subcommand) {
+		case 'get': return getPreference(settings, key, opts);
+		case 'set': return setPreference(settings, key, value, opts);
+		case 'reset':
+		case 'delete': return resetPreference(settings, key, opts);
+		case 'enable': return enablePreference(settings, key, opts);
+		case 'disable': return disablePreference(settings, key, opts);
+	}
+}
+
+interface SubcommandOptions {
+	location?: 'global' | 'project';
+	json?: boolean;
+}
+
+// Default `location` to "project" to avoid reading default preferencesa
+async function getPreference(settings: AstroSettings, key: PreferenceKey, { location = 'project' }: SubcommandOptions) {
+	try {
+		const value = await settings.preferences.get(key, { location });
+		// TODO: guard against printing objects
+		if (value !== undefined) {
+			console.log(msg.preferenceGet(key, value));
+		} else {
+			const defaultValue = await settings.preferences.get(key);
+			console.log(msg.preferenceDefault(key, defaultValue));
+		}
+		return 0;
+	} catch {}
+	return 1;
+}
+
+async function setPreference(settings: AstroSettings, key: PreferenceKey, value: unknown, { location }: SubcommandOptions) {
+	try {
+		await settings.preferences.set(key, value as any, { location });
+		console.log(msg.preferenceSet(key, value))
+		return 0;
+	} catch {}
+	return 1;
+}
+
+async function enablePreference(settings: AstroSettings, key: PreferenceKey, { location }: SubcommandOptions) {
+	try {
+		await settings.preferences.set(key, true, { location });
+		console.log(msg.preferenceEnabled(key.replace('.enabled', '')))
+		return 0;
+	} catch {}
+	return 1;
+}
+
+async function disablePreference(settings: AstroSettings, key: PreferenceKey, { location }: SubcommandOptions) {
+	try {
+		await settings.preferences.set(key, false, { location });
+		console.log(msg.preferenceDisabled(key.replace('.enabled', '')))
+		return 0;
+	} catch {}
+	return 1;
+}
+
+async function resetPreference(settings: AstroSettings, key: PreferenceKey, { location }: SubcommandOptions) {
+	try {
+		await settings.preferences.set(key, undefined as any, { location });
+		console.log(msg.preferenceReset(key))
+		return 0;
+	} catch {}
+	return 1;
+}
+
+async function listPreferences(settings: AstroSettings, { location, json }: SubcommandOptions) {
+	const store = await settings.preferences.getAll({ location });
+	if (json) {
+		console.log(JSON.stringify(store, null, 2));
+	}
+	// TODO: pretty print
+	console.log(JSON.stringify(store, null, 2));
```

Comment:
Should this be an else-if?

Reply:
Whoops, yes. I think I meant to `return` in the `if`.